vms/avm/index_test.go

```diff
@@ -5,349 +5,177 @@ package avm
-	"github.com/ava-labs/avalanchego/snow/engine/common"
+	"github.com/ava-labs/avalanchego/utils"
 	"github.com/ava-labs/avalanchego/utils/constants"
 	"github.com/ava-labs/avalanchego/utils/crypto/secp256k1"
-	"github.com/ava-labs/avalanchego/utils/wrappers"
-	"github.com/ava-labs/avalanchego/version"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/avalanchego/vms/avm/config"
 	"github.com/ava-labs/avalanchego/vms/avm/txs"
 	"github.com/ava-labs/avalanchego/vms/components/avax"
 	"github.com/ava-labs/avalanchego/vms/components/index"
 	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
 )
 
-var indexEnabledAvmConfig = Config{
-	IndexTransactions: true,
-}
-
 func TestIndexTransaction_Ordered(t *testing.T) {
 	require := require.New(t)
 
-	genesisBytes := BuildGenesisTest(t)
-	issuer := make(chan common.Message, 1)
-	baseDBManager := manager.NewMemDB(version.Semantic1_0_0)
-	ctx := NewContext(t)
-	genesisTx := GetAVAXTxFromGenesisTest(genesisBytes, t)
-	avaxID := genesisTx.ID()
-	vm := setupTestVM(t, ctx, baseDBManager, genesisBytes, issuer, indexEnabledAvmConfig)
+	env := setup(t, &envConfig{
+		vmStaticConfig: &config.Config{},
+	})
 	defer func() {
-		require.NoError(vm.Shutdown(context.Background()))
-		ctx.Lock.Unlock()
+		require.NoError(env.vm.Shutdown(context.Background()))
+		env.vm.ctx.Lock.Unlock()
 	}()
 
 	key := keys[0]
 	addr := key.PublicKey().Address()
+	txAssetID := avax.Asset{ID: env.genesisTx.ID()}
 
-	var uniqueTxs []*UniqueTx
-	txAssetID := avax.Asset{ID: avaxID}
-
-	ctx.Lock.Lock()
+	var txs []*txs.Tx
 	for i := 0; i < 5; i++ {
-		// create utxoID and assetIDs
+		// make utxo
 		utxoID := avax.UTXOID{
 			TxID: ids.GenerateTestID(),
 		}
+		utxo := buildUTXO(utxoID, txAssetID, addr)
+		env.vm.state.AddUTXO(utxo)
 
-		// build the transaction
+		// make transaction
 		tx := buildTX(utxoID, txAssetID, addr)
-
-		// sign the transaction
-		require.NoError(signTX(vm.parser.Codec(), tx, key))
-
-		// Provide the platform UTXO
-		utxo := buildPlatformUTXO(utxoID, txAssetID, addr)
-
-		// save utxo to state
-		vm.state.AddUTXO(utxo)
+		require.NoError(tx.SignSECP256K1Fx(env.vm.parser.Codec(), [][]*secp256k1.PrivateKey{{key}}))
 
 		// issue transaction
-		_, err := vm.IssueTx(tx.Bytes())
-		require.NoError(err)
-
-		ctx.Lock.Unlock()
-		require.Equal(common.PendingTxs, <-issuer)
-		ctx.Lock.Lock()
-
-		// get pending transactions
-		txs := vm.PendingTxs(context.Background())
-		require.Len(txs, 1)
-		parsedTx := txs[0]
-		uniqueParsedTX := parsedTx.(*UniqueTx)
-		uniqueTxs = append(uniqueTxs, uniqueParsedTX)
-
-		var inputUTXOs []*avax.UTXO
-		for _, utxoID := range uniqueParsedTX.InputUTXOs() {
-			utxo, err := vm.state.GetUTXO(utxoID.InputID())
-			require.NoError(err)
-
-			inputUTXOs = append(inputUTXOs, utxo)
-		}
+		issueAndAccept(require, env.vm, env.issuer, tx)
 
-		// index the transaction
-		require.NoError(vm.addressTxsIndexer.Accept(uniqueParsedTX.ID(), inputUTXOs, uniqueParsedTX.UTXOs()))
```

Comment:
I have no clue why we didn't actually `Accept` the transaction previously to test the indexing flow... instead we manually indexed it... Now we just accept it normally.